servo/types.py

```diff
@@ -1253,6 +1253,10 @@ class files into memory, just-in-time compilation being appliied to critical
     semantics.
     """
 
+    environment: Optional[Dict[str, Any]] = None
```

Comment:
Can you model this with a more explicit typing? Environment should be its own model with a "mode" field of type str

Reply:
TBH, I would not have modeled 'environment' at all and allowed it (and other extras) to be added and propagated to event handlers - rather than being silently dropped with no warning or notice, as it does now. I added the 'str->any' data type just because it was the easiest change (I know there has to be a way in Pydantic to add 'any extra item allowed', but I haven't bothered finding it).

This may be useful for 'experimental' features (that can be added as connectors, similar to the 'https://github.com/opsani/servo-live-app-emulator' add-on connector) - without requiring a change in the core ServoX code for every tiny little addition to the data.